UnityProject/Assets/Scripts/Construction/Machine.cs

```diff
@@ -100,8 +100,14 @@ public void WhenDestroyed(DestructionInfo info)
 				itemStorage.ServerDropAll();
 			}
 
-			var frame = Spawn.ServerPrefab(framePrefab, SpawnDestination.At(gameObject)).GameObject;
-
+			SpawnResult frameSpawn = Spawn.ServerPrefab(framePrefab, SpawnDestination.At(gameObject));
+			if (!frameSpawn.Successful)
+			{
+				throw new MissingReferenceException(
```

Comment:
is it safe for us to throw exceptions? Maybe using the debug function for errors is prefered.

Reply:
Yeah better to do a 
`Debug.LogError("words");
return;
`